src/CattyTests/DataModel/UserListTest.swift

```diff
@@ -0,0 +1,74 @@
+/**
+ *  Copyright (C) 2010-2020 The Catrobat Team
+ *  (http://developer.catrobat.org/credits)
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  An additional term exception under section 7 of the GNU Affero
+ *  General Public License, version 3, is available at
+ *  (http://developer.catrobat.org/license_additional_term)
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see http://www.gnu.org/licenses/.
+ */
+
+import XCTest
+
+@testable import Pocket_Code
+
+final class UserListTest: XCTestCase {
+
+    override func setUp() {
+        super.setUp()
+    }
+
+    func testInit() {
+        let userList1 = UserList(name: "testList")
+        let userList2 = UserList(list: userList1)
+
+        XCTAssertTrue(userList1.isEqual(userList2))
+        XCTAssertFalse(userList1 === userList2)
+    }
+
+    func testMutableCopyWithContext() {
+        let userList = UserList(name: "testList")
+
+        let context = CBMutableCopyContext()
+        XCTAssertEqual(0, context.updatedReferences.count)
+
+        let userListCopy = userList.mutableCopy(with: context) as! UserList
+        XCTAssertEqual(userList.name, userListCopy.name)
+        XCTAssertTrue(userList === userListCopy)
+    }
+
+    func testMutableCopyAndUpdateReference() {
+        let userListA = UserList(name: "userList")
+        let userListB = UserList(name: "userList")
+
+        let context = CBMutableCopyContext()
+        context.updateReference(userListA, withReference: userListB)
+        XCTAssertEqual(1, context.updatedReferences.count)
+
+        let userListCopy = userListA.mutableCopy(with: context) as! UserList
+        XCTAssertEqual(userListA.name, userListCopy.name)
+        XCTAssertFalse(userListA === userListCopy)
+        XCTAssertTrue(userListB === userListCopy)
+    }
+
+    func testIsEqualToUserListForDifferentName() {
```

Comment:
Please also consider the actual values in your tests (like in https://github.com/Catrobat/Catty/blob/develop/src/CattyTests/DataModel/UserVariableTests.swift#L94).

Reply:
Hi @m-herold  `UserList.value` should always be a list right? how to assign `NSMutableArray` to the value of a `UserList` which is of type `[Any]?`. Also if the value is not assigned an `NSMutableArray` then the `Util.Equal` function does not work

please correct me if I am doing something wrong.

Reply:
Yes, this value should always be an array. Ok, that would mean we will have to write our own comparison function for two arrays containing either Strings or numbers. For that, I will create another ticket, please do not do this as part of this ticket. Instead, for now, change the type to _NSMutableArray_ (do not forget to add the _typealias_). Please note: change optional to non-optional, and initialize an empty array in the initializer. You will have to make one minor change to the protocol for that and add an optional _typealias_ for the _UserVariable_. Thanks!